### PR TITLE
Add api tests

### PR DESCRIPTION
**Pull request by awgreene**
## Proposed changes

Added a few tests to the API Server. Will add more, but please feel free to add comments.

## Checklist
- [x] I have read the [CONTRIBUTING](https://github.com/mauricioabreu/go-hangman/blob/master/CONTRIBUTING.md) doc
- [x] Lint and unit tests pass locally with my changes (if appropriate)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: http/api_test.go [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+	}
+
+	// Delete game.
+	gameID := strings.Split(w.HeaderMap["Location"][0], "/")[2]
+
+	r, _ = http.NewRequest("GET", "/games/"+gameID, nil)
+	w = httptest.NewRecorder()
+	router.ServeHTTP(w, r)
+	if w.Code != 200 {
+		t.Errorf("Failed to delete game %v", w)
```
Comment (mauricioabreu) [inline http/api_test.go]: I think you meant `get`

File: http/api_test.go [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+func TestCreateAndGetGame(t *testing.T) {
+	// Create game.
+	r, _ := http.NewRequest("POST", "/games", nil)
+	w := httptest.NewRecorder()
+	router.ServeHTTP(w, r)
+	if w.Code != 204 {
+		t.Error("Failed to create new game.", w.Code)
+	}
+
+	// Delete game.
```
Comment (mauricioabreu) [inline http/api_test.go]: Here as well: `get game`?

**Review by mauricioabreu — APPROVED** [review]
Minor things to fix

**Comment by mauricioabreu** [comment]
Good job! You covered most of the features with a few and simple tests. Thank you!

**Comment by awgreene** [comment]
Thanks @mauricioabreu, as it stands the api tests cover 72.4% of statements. The remaining statements seem to be related to the `main()` function or error checks. Hope this helps!

**Comment by mauricioabreu** [comment]
Thank you very much @awgreene it was a very good help.
One question: are those files ok? It looks like go-hangman was added to the vendor path. Can you review it?

**Comment by awgreene** [comment]
@mauricioabreu removed the vendor stuff.

**Review by mauricioabreu — APPROVED** [review]
Thank you sooo much! Nice code.

**Pull request merged**